lib/ansible/module_utils/docker_common.py

```diff
@@ -410,12 +411,20 @@ def _image_lookup(self, name, tag):
         exists.
         '''
         try:
-            response = self.images(name=name)
+            if name in DOCKER_RESERVED_IMAGE_NAMES:
+                response = self.images()
+            else:
+                response = self.images(name=name)
         except Exception as exc:
             self.fail("Error searching for image %s - %s" % (name, str(exc)))
         images = response
-        if tag:
-            lookup = "%s:%s" % (name, tag)
+        lookup = "%s:%s" % (name, tag)
+        if tag and name in DOCKER_RESERVED_IMAGE_NAMES:
+            for image in response:
+                if image['Id'] == lookup:
```

Comment:
> I feel like a [WARNING] should pop up when you do that

When specify a hash string that matches more than one image? We should `fail` immediately, this is users' fault.